op-dispute-mon/mon/monitor_test.go

```diff
@@ -0,0 +1,248 @@
+		require.NoError(t, err)
+		require.Equal(t, 1, status.calls)
+	})
+
+	t.Run("Record status success", func(t *testing.T) {
+		monitor, source, metrics, _ := setupMonitorTest(t)
+		source.games = []types.GameMetadata{{}, {}, {}}
+		err := monitor.monitorGames(context.Background())
+		require.NoError(t, err)
+		require.Equal(t, len(source.games), metrics.inProgress)
+	})
+}
+
+func TestMonitorGames(t *testing.T) {
+	t.Run("Monitors games", func(t *testing.T) {
+		addr1 := common.Address{0xaa}
+		addr2 := common.Address{0xbb}
+		monitor, source, metrics, _ := setupMonitorTest(t)
+		source.games = []types.GameMetadata{newFDG(addr1, 9999), newFDG(addr2, 9999)}
+		source.maxSuccess = 2 // Only allow two successful fetches
+
+		monitor.StartMonitoring()
+		require.Eventually(t, func() bool {
+			return metrics.inProgress == 2
+		}, time.Second, 100*time.Millisecond)
+		monitor.StopMonitoring()
+		require.Equal(t, source.maxSuccess*len(source.games), metrics.inProgress) // Each game's status is recorded twice
+	})
+
+	t.Run("Fails to monitor games", func(t *testing.T) {
+		monitor, source, metrics, _ := setupMonitorTest(t)
+		source.fetchErr = errors.New("boom")
+
+		monitor.StartMonitoring()
+		require.Eventually(t, func() bool {
+			return source.calls > 0
+		}, time.Second, 100*time.Millisecond)
+		monitor.StopMonitoring()
+		require.Equal(t, 0, metrics.inProgress)
+		require.Equal(t, 0, metrics.defenderWon)
+		require.Equal(t, 0, metrics.challengerWon)
+	})
+}
+
+func newFDG(proxy common.Address, timestamp uint64) types.GameMetadata {
+	return types.GameMetadata{
+		Proxy:     proxy,
+		Timestamp: timestamp,
+	}
+}
+
+func setupMonitorTest(t *testing.T) (*gameMonitor, *stubGameSource, *stubMonitorMetricer, *mockStatusLoader) {
+	logger := testlog.Logger(t, log.LvlDebug)
+	source := &stubGameSource{}
+	i := uint64(1)
+	fetchBlockNum := func(ctx context.Context) (uint64, error) {
+		i++
+		return i, nil
+	}
+	fetchBlockHash := func(ctx context.Context, number *big.Int) (common.Hash, error) {
+		return common.Hash{}, nil
+	}
+	metrics := &stubMonitorMetricer{}
+	monitorInterval := time.Duration(100 * time.Millisecond)
+	status := &mockStatusLoader{}
+	monitor := newGameMonitor(
+		logger,
+		metrics,
+		clock.NewSimpleClock(),
+		monitorInterval,
+		source,
+		status,
+		time.Duration(10*time.Second),
+		fetchBlockNum,
+		fetchBlockHash,
+	)
+	return monitor, source, metrics, status
+}
+
+type mockStatusLoader struct {
+	calls  int
+	status types.GameStatus
+	err    error
+}
+
+func (m *mockStatusLoader) GetStatus(ctx context.Context, _ common.Address) (types.GameStatus, error) {
+	m.calls++
+	if m.err != nil {
+		return 0, m.err
+	}
+	return m.status, nil
+}
+
+type stubMonitorMetricer struct {
+	inProgress    int
+	defenderWon   int
+	challengerWon int
+}
+
+func (s *stubMonitorMetricer) RecordGamesStatus(inProgress, defenderWon, challengerWon int) {
+	s.inProgress += inProgress
+	s.defenderWon += defenderWon
+	s.challengerWon += challengerWon
+}
```

Comment:
This doesn't match what the actual metrics will do.  And incrementing like this will wind up counting the same game multiple times - once for each time the monitor ticker fires.